File: 007_Bitwise_Operators_Modifiers/1_data.py
# Use int for small values like age (Python auto-handles size)
age = 25
print("Age (int):", age)

# Use int for larger numbers like population or file size
population = 1_000_000
print("Population (int):", population)

# Use int for very large numbers (Python handles big integers automatically)
stars_in_galaxy = 9223372036854775807
print("Stars in Galaxy (int):", stars_in_galaxy)

# Use int for signed values (positive/negative), e.g., temperature
temperature = -15
print("Temperature (int):", temperature, "°C")

# Use int for unsigned-like positive values, e.g., student count
student_count = 150
print("Student Count (int):", student_count)




# In Python, int automatically behaves like short, long, and long long — there's no need to specify size.

# There's no unsigned or signed keyword; Python int can represent both positive and negative numbers by default.

# Python automatically allocates memory as needed (unlike C++ fixed-size integers).

File: 007_Bitwise_Operators_Modifiers/1_data.cpp
#include <iostream>
using namespace std;

int main() {
    // Use short when value is small, e.g., age (saves memory)
    short age = 25;
    cout << "Age (short): " << age << endl;

    // Use long for larger numbers like population or file sizes
    long population = 1000000L;
    cout << "Population (long): " << population << endl;

    // Use long long for very large numbers like scientific data
    long long starsInGalaxy = 9223372036854775807LL;
    cout << "Stars in Galaxy (long long): " << starsInGalaxy << endl;

    // Use signed when values can be both positive and negative, e.g., temperature
    signed int temperature = -15;
    cout << "Temperature (signed int): " << temperature << "°C" << endl;

    // Use unsigned when values are always positive, e.g., count or ID
    unsigned int studentCount = 150;
    cout << "Student Count (unsigned int): " << studentCount << endl;

    return 0;
}



// Age (short): 25
// Population (long): 1000000
// Stars in Galaxy (long long): 9223372036854775807
// Temperature (signed int): -15°C
// Student Count (unsigned int): 150